app/models/work_activity.rb

```diff
@@ -29,6 +34,22 @@ def self.add_system_activity(work_id, message, user_id, activity_type: "SYSTEM",
     activity
   end
 
+  def self.activities_for_work(work_id, types = [])
+    context = where(work_id: work_id).order(updated_at: :desc)
+    if types.count > 0
+      context = context.where(activity_type: types)
```

Comment:
Instead of doing a select after the fact as the old code did add it to the where clause to let the database do what it does best.